testing/idris/APKBUILD

```diff
@@ -0,0 +1,116 @@
+# Contributor: Mitch Tishmack <mitch.tishmack@gmail.com>
+# Contributor: Jakub Jirutka <jakub@jirutka.cz>
+# Maintainer: Mitch Tishmack <mitch.tishmack@gmail.com>
+#
+# Note: This abuilds downloads Haskell dependencies in the prepare phase
+#       without any verification!
+pkgname=idris
+pkgver=1.0
+pkgrel=0
+pkgdesc="A Language with Dependent Types"
+url="http://www.idris-lang.org"
+arch="x86_64 armhf"
+license="BSD3"
+depends="gmp-dev gcc"
+makedepends="ghc cabal libffi-dev ncurses-dev zlib-dev"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/idris-lang/Idris-dev/archive/v$pkgver.tar.gz
+	cabal.config"
+builddir="$srcdir/Idris-dev-$pkgver"
+
+export CABAL_CONFIG="$srcdir/cabal/cabal.config"
+
+cabal_update() {
+	msg "Updating constraints"
+
+	cd "$builddir"
+
+	# Create cabal config and fetch index.
+	HOME="${CABAL_CONFIG%/*}" cabal update
+
+	# Resolve deps and generate fresh cabal.config with version constraints.
+	rm -f cabal.config
+	cabal freeze
+
+	# Add version tag at the first line.
+	sed -i "1i--$pkgver" cabal.config
+
+	mv cabal.config "$startdir"/
+
+	cd "$startdir"
+	abuild checksum
+}
+
+prepare() {
+	default_prepare
+
+	if [ "$(head -n 1 "$srcdir"/cabal.config)" != "--$pkgver" ]; then
+		die "Requirements file is outdated, run 'abuild cabal_update'"
+	fi
+
+	cd "$builddir"
+
+	# Create cabal config and fetch index.
+	HOME="${CABAL_CONFIG%/*}" cabal update
+
+	# Cabal implicitly loads cabal.config from the project's directory.
+	ln -sf "$srcdir"/cabal.config .
+
+	# Create Cabal sandbox that will be used for all subsequent cabal
+	# invocations. This is something like Python virtual env.
+	cabal sandbox init
+
+	# Fetch all dependencies.
+	# XXX: This should always fetch the same versions as defined in
+	# our cabal.config, but Cabal does not verify packages integrity!
```

Comment:
This is incorrect, hackage and cabal implement TUF The Update Framework.

https://www.well-typed.com/blog/2015/04/improving-hackage-security/

Unless what you're referring to here via package integrity is comparing a sha sum or digest, in which case yes that isn't directly performed. Though obliquely in that if the package changes, it will no longer pass verification due to not having been signed by hackage at both the index level and/or the package level.